subsys/usb/device_next/class/usbd_msc.c

```diff
@@ -724,30 +733,43 @@ static int msc_bot_request_handler(struct usbd_class_node *const node,
 }
 
 /* Class associated configuration is selected */
-static void msc_bot_enable(struct usbd_class_node *const node)
+static void msc_bot_enable(struct usbd_class_data *const c_data)
 {
-	struct msc_bot_ctx *ctx = node->data->priv;
+	struct msc_bot_ctx *ctx = usbd_class_get_private(c_data);
 
 	LOG_INF("Enable");
 	atomic_set_bit(&ctx->bits, MSC_CLASS_ENABLED);
-	msc_bot_schedule_reset(node);
+	msc_bot_schedule_reset(c_data);
 }
 
 /* Class associated configuration is disabled */
-static void msc_bot_disable(struct usbd_class_node *const node)
+static void msc_bot_disable(struct usbd_class_data *const c_data)
 {
-	struct msc_bot_ctx *ctx = node->data->priv;
+	struct msc_bot_ctx *ctx = usbd_class_get_private(c_data);
 
 	LOG_INF("Disable");
 	atomic_clear_bit(&ctx->bits, MSC_CLASS_ENABLED);
 }
 
+static void *msc_bot_get_desc(struct usbd_class_data *const c_data,
+			      const enum usbd_speed speed)
+{
+	struct msc_bot_ctx *ctx = usbd_class_get_private(c_data);
+
+	if (speed == USBD_SPEED_HS) {
+		return ctx->hs_desc;
+	}
+
+	return ctx->fs_desc;
+}
+
 /* Initialization of the class implementation */
-static int msc_bot_init(struct usbd_class_node *const node)
+static int msc_bot_init(struct usbd_class_data *const c_data)
 {
-	struct msc_bot_ctx *ctx = node->data->priv;
+	struct msc_bot_ctx *ctx = usbd_class_get_private(c_data);
+	struct msc_bot_desc *desc = ctx->desc;
```

Comment:
```suggestion
```

This fails build due to `usbd_msc.c:770:30: error: unused variable 'desc' [-Werror=unused-variable]`